### ISSUE TITLE
Writes to the same address without an erase

### ISSUE THREAD

**Issue by clivebluston** [pellepl/niffs#6]
I made some experiment with the code and found that there are writes to the same address without an erase in between. This would cause the NOR flash drivers that I work with to return an error.
WRIT     **0x638866** : 122
WRIT     0x638862 : 2
WRIT     0x638756 : 2
WRIT     **0x638866** : 4
WRIT     0x638356 : 2
WRIT     0x6387e4 : 2
WRIT     0x638862 : 2

I also see that there has been no activity for a while.
So I was wondering what is the status of the code?
Thanks

**Comment by pellepl** [comment]
Niffs does write data to already written addresses for status updates (e.g. deleted pages). If it is possible, you'd need to rewrite your NOR flash driver. If that is not an option, I believe niffs can be patched to accommodate no writes on previously written bytes, but it would need some effort.

As for the , I haven't done anything with the code for some year(s), but it still serves the purpose I wrote it in the first hand - and is still running in that project. I simply haven't had any reason to change things. :) That said, if you find a bug please report it.

**Comment by clivebluston** [comment]
I also noticed there is no license. This would be an issue contributing to the decision to use it.
Is it an oversight?

**Comment by pellepl** [comment]
Yes, pure oversight. As a matter of fact I promised to add one months ago.
Hold on.

2017-05-17 13:45 GMT+02:00 Clive Bluston <notifications@github.com>:

**Comment by pellepl** [comment]
... and there it is.

**Comment by clivebluston** [comment]
Thanks, that is great!
I know that theretically you should be able to flip 1's that havent yet been flipped, however I played around with the driver by removing the code protecting a subsequent write without an erase, but the hardware operation failed. So I think it is something that I may have to live with. I will take a look at your code to try and understand how it can be patched.
Thanks for your help.

**Comment by RallyTronics** [comment]
There are a number of micro-controllers that don't allow you to write the flash multiple times without erasing.
